### ISSUE TITLE
[Bug]: SpinButton incorrectly handles invalid data if the app doesnt switch focus on increment/decrement buttons 

### ISSUE THREAD

**Issue by thomabr** [microsoft/fluentui#23118]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
shell
System:
OS: Windows 10 10.0.19043
CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
Memory: 5.75 GB / 15.91 GB
Browsers:
Edge: Spartan (44.19041.1266.0), Chromium (100.0.1185.39)
Internet Explorer: 11.0.19041.1566
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://codepen.io/thomabr/pen/yLpGvdw

## Actual Behavior
Type in an invalid number "1 1" into the spin button
Hit increment button
click elsewhere to find focus.
The value is constantly incrementing

## Expected Behavior
Increment button validates the value and returns it to the last known good.

Office likes to not lose focus on elements when buttons are pressed in our UI. This allows dual mouse/keyboard usage. However, it seems because we don't lose focus from the input element, validation is never called, but is assumed to be present later on

I think we can fix this by adding a mouse down event as well here
https://github.com/microsoft/fluentui/blob/c4e51d88b0be5acd7d21882c8a448c64c2645080/packages/react/src/components/SpinButton/SpinButton.base.tsx#L251

```
 if (ev.type === 'keydown' || ev.type === 'mousedown') {
          validate(ev);
        }
```

I believe this should fix it but would like the team to understand the repercussions here 

Reopening 
https://github.com/microsoft/fluentui/issues/22532

### Requested priority

Normal

### Products/sites affected

Office Web Apps

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
@thomabr - Thanks for filing this issue with us. To set expectations the developers will review this issue once capacity allows.
@khmakoto - Would you be able to confirm if this is a regression, or if this behavior is an issue?

**Comment by spmonahan** [comment]
@thomabr I just merged a pull request with a fix for this. Rather than have the value revert to the last known good value, the value is stepped from the last known good value. I left the implementation this way because when following your repro steps but using the arrow keys rather than clicking the buttons this is the current behavior so I figured it was best to keep that for button clicks.

Thank you for the bug report and for pointing out where the issue was in source. Contributions like these are always really helpful!
